docs/golang/README.md

```diff
@@ -136,21 +136,43 @@ The `graph` package goes through all the `type:contact` messages and just stores
 
 The `plugins2/names` package implements the same muxrpc interface as [ssb-names](https://github.com/ssbc/ssb-names). It does a similar thing to `graph` but for `type:about` messages and the name, image and description information in them.
 
-### MultiLog
+### [`margaret/multilog`](https://pkg.go.dev/go.cryptoscope.co/margaret)
 
-We wanted to have a unified way to access different kinds of _sub logs_ (of the big receive log, for instance).
+We wanted to have a unified way to access different kinds of sublogs, that is, logs-as-slices into a larger log. One example: accessing parts of the big receive log (the posts by one particular feed, for instance).
 
-Enter [MultiLog](https://pkg.go.dev/go.cryptoscope.co/margaret/multilog), it returns `margaret.Log`s for specific addresses.
+Enter [MultiLog](https://pkg.go.dev/go.cryptoscope.co/margaret/multilog), which returns `margaret.Log`s for specific addresses.
 
-#### Implementation details
+Package `multilog` is one of a few ways that go-ssb creates an overarching structure (indexes and views) of the
+messages being received from all over the scuttleverse. Another prominent indexing solution is
+the [`margaret/indexes`](https://godocs.io/go.cryptoscope.co/margaret/indexes) package, which
+more or less functions as a key-value store. One of the differences between `margaret/indexes` and
+`margaret/multilog`: the former stores arbitrary data (think: strings of your choosing), while
+the latter only stores ssb messages.
 
-The first attempt we made stored an entry like `@publicKey:N -> ReceiveSeqM` (where N is the logical sequence entry on the feed and M the sequence of the message in the receive log) in a key-value database (`margaret/multilog/badger`). This worked but came with a lot of churn since the values are quite small, additionally keys are large and redundant.
+#### A `multilog` explanation
 
-A newer approach is to store the set of receivelog sequences as a special kind of [Bitmap](https://en.wikipedia.org/wiki/Bitmap). (tl;dr: You can imagine these as a compressed array of integers.) This is advantageous not only because we only store one bitmap for all the messages in a set (like `by author:x` or `by type:y`) but also because it allows us to use them as compound indexes since they can be logically combined via boolean algebra (`x AND y` gives us the intersection, `x OR y` gives us the union) since they all map to the same messages in the receive log.
+One way to think about the `multilog` package: 
+
+A multilog can be seen as a _[tree](https://en.wikipedia.org/wiki/Tree_(data_structure))-like_
+index. The multilog itself is a log—or the root of the tree—which leads to other logs. These
+other logs are called _sublogs_, and each sublog has many leaves. Each leaf corresponds to some
+message. What type of message is stored depends on the particular type of data being indexed by the multilog in question (all follow messages by `<peer>`, for example).
```

Comment:
```suggestion
message. What type of message is stored depends on the particular type of data being indexed by the multilog in question (i.e. all `type:contact` messages, all messages by `<peer>`, etc.).
```

Reply:
the example is not directly indexed but could be constructed by boolean combination of two indexes.